site/app/controllers/AutogradingStatusController.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace app\controllers;
+
+use app\libraries\Core;
+use app\libraries\response\MultiResponse;
+use app\libraries\response\WebResponse;
+use app\libraries\response\JsonResponse;
+use app\libraries\routers\AccessControl;
+use app\libraries\GradingQueue;
+use app\views\AutogradingStatusView;
+use app\views\ErrorView;
+use Symfony\Component\Routing\Annotation\Route;
+
+class AutogradingStatusController extends AbstractController {
```

Comment:
As far as I know,`@AccessControl` only checks your access for the current course. However, this page doesn't exist as a course page so `AccessControl` would not work.

Reply:
Hmm, ok.  I'm still a little wary of this because someone could easily come in later and make changes, not knowing that access control had to be done in each function, and unknowingly leave a security vulnerability open.  It's tough though so if there's no better option I'll approve it the way it is.

Reply:
Yeah I agree it isn't the greatest solution. I should also note that something similar is also done in the docker UI page, not to justify this but as a note of where else this is happening.

Reply:
Maybe just add a comment saying this is a security check. I have seen other places in the code do that when a manual security check was needed.